extension-style/src/main/java/com/mapbox/maps/extension/style/layers/Layer.kt

```diff
@@ -111,7 +113,8 @@ abstract class Layer : StyleContract.StyleLayerExtension {
   override fun bindTo(delegate: StyleInterface, position: LayerPosition?) {
     this.delegate = delegate
 
-    val expected = delegate.addStyleLayer(getCachedLayerProperties(), position)
+    val propertiesValue = appliedLayerPropertiesValue ?: getCachedLayerProperties()
+    val expected = delegate.addStyleLayer(propertiesValue, position)
```

Comment:
once set the `appliedLayerPropertiesValue`, we should also be able to set the runtime-cached property values in the `getCachedLayerProperties`, i.e.
```
val properties = layerProperties.values.filter { it.propertyName != "id" &&  it.propertyName != "type" && it.propertyName != "source"}
properties.forEach { 
  delegate?.setStyleLayerProperty(layerId, it.propertyName, it.value)
}
```

Reply:
done